File: backend/src/controllers/kpi.controller.js
import { kpiGlobal, kpiPorCategoria } from '../services/kpi.service.js';

export async function obtenerKPIs(_req, res) {
  try {
    const [global, categorias] = await Promise.all([kpiGlobal(), kpiPorCategoria()]);
    res.json({ global, categorias });
  } catch (err) {
    console.error(err);
    res.status(500).json({ error: 'No se pudieron calcular las KPIs' });
  }
}

File: backend_dart_kpis/lib/controllers/kpi_controller.dart
import 'dart:convert';
import 'package:shelf/shelf.dart';
import '../services/kpi_service.dart';

Future<Response> obtenerKPIsController(Request req) async {
  try {
    final kpis = await obtenerKPIs();
    return Response.ok(jsonEncode(kpis),
        headers: {'Content-Type': 'application/json'});
  } catch (e) {
    return Response.internalServerError(
      body: jsonEncode({'error': 'No se pudieron calcular las KPIs'}),
      headers: {'Content-Type': 'application/json'},
    );
  }
}
